pkg/container/docker/client.go

```diff
@@ -203,18 +210,282 @@ func setupPortBindings(hostConfig *container.HostConfig, portBindings map[string
+	}
+
+	// Create container options
+	config := &container.Config{
+		Image:        EgressImage,
+		Cmd:          nil,
+		Env:          nil,
+		Labels:       egressLabels,
+		AttachStdin:  attachStdio,
+		AttachStdout: attachStdio,
+		AttachStderr: attachStdio,
+		OpenStdin:    attachStdio,
+		Tty:          false,
+	}
+
+	configDns := &container.Config{
+		Image:        DnsImage,
+		Cmd:          nil,
+		Env:          nil,
+		Labels:       dnsLabels,
+		AttachStdin:  attachStdio,
+		AttachStdout: attachStdio,
+		AttachStderr: attachStdio,
+		OpenStdin:    attachStdio,
+		Tty:          false,
+	}
+
+	// generate the squid configuration and mount it
+	squidConfPath, err := createTempSquidConf(perm, containerName, exposedPorts)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to create temporary squid.conf: %v", err)
+	}
+
+	mounts := []runtime.Mount{}
+	mounts = append(mounts, runtime.Mount{
+		Source:   squidConfPath,
+		Target:   "/etc/squid/squid.conf",
+		ReadOnly: true,
+	})
+
+	// Create egress host configuration
+	egressHostConfig := &container.HostConfig{
+		Mounts:      convertMounts(mounts),
+		NetworkMode: container.NetworkMode("bridge"),
+		CapAdd:      []string{"CAP_SETUID", "CAP_SETGID"},
+		CapDrop:     nil,
+		SecurityOpt: nil,
+		RestartPolicy: container.RestartPolicy{
+			Name: "unless-stopped",
+		},
+	}
+	dnsHostConfig := &container.HostConfig{
+		Mounts:      convertMounts(mounts),
+		NetworkMode: container.NetworkMode("bridge"),
+		CapAdd:      nil,
+		CapDrop:     nil,
+		SecurityOpt: nil,
+		RestartPolicy: container.RestartPolicy{
+			Name: "unless-stopped",
+		},
+	}
+
+	// create networks
+	networkName := fmt.Sprintf("toolhive-%s-internal", containerName)
+	endpointsConfig := map[string]*network.EndpointSettings{
+		networkName:         {},
+		"toolhive-external": {},
+	}
+
+	// Setup port bindings
+	if err := setupExposedPorts(config, exposedPorts); err != nil {
+		return "", "", NewContainerError(err, "", err.Error())
+	}
+	if err := setupPortBindings(egressHostConfig, portBindings); err != nil {
+		return "", "", NewContainerError(err, "", err.Error())
+	}
+
+	// Create egress container itself
+	egressContainerId, err := c.createContainer(ctx, egressContainerName, config, egressHostConfig, endpointsConfig)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to create egress container: %v", err)
+	}
+
+	// now create the dns container
+	dnsContainerId, err := c.createContainer(ctx, dnsContainerName, configDns, dnsHostConfig, endpointsConfig)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to create dns container: %v", err)
+	}
+
+	return egressContainerId, dnsContainerId, nil
+}
+
 // DeployWorkload creates and starts a workload.
 // It configures the workload based on the provided permission profile and transport type.
 // If options is nil, default options will be used.
-func (c *Client) DeployWorkload(
+func (c *Client) DeployWorkload( // nolint:gocyclo
```

Comment:
In general, we should avoid adding in these nolint directives if they are avoidable, perhaps factoring some of the networking code in a separate method would help here? (and help keep this function easy to read)